src/pal/src/file/file.cpp

```diff
@@ -3605,18 +3721,26 @@ GetTempFileNameW(
             goto done;
         }
     }
-       
+    
+    tempfile_name = (CHAR *)alloca((MAX_LONGPATH+1) * sizeof(CHAR));
```

Comment:
alloca doesn't allocate on heap, it reserves space on stack. So if we always want to allocate MAX_LONGPATH, you can leave it the way it was before - array local variable.